drivers/i2c/i2c_ll_stm32.c

```diff
@@ -23,12 +23,21 @@ static int i2c_stm32_runtime_configure(struct device *dev, u32_t config)
 	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
 	struct i2c_stm32_data *data = DEV_DATA(dev);
 	I2C_TypeDef *i2c = cfg->i2c;
-	u32_t clock;
+	u32_t clock = 0;
 
-	data->dev_config = config;
+#if defined(CONFIG_SOC_SERIES_STM32F3X) || defined(CONFIG_SOC_SERIES_STM32F0X)
+	struct stm32_pclken pclken = {
+		.bus = STM32_CLOCK_BUS_AHB2
+	};
 
 	clock_control_get_rate(device_get_binding(STM32_CLOCK_CONTROL_NAME),
+			(clock_control_subsys_t *) &pclken, &clock);
```

Comment:
I'm affraid this won't work.
With this you'll get SystemCoreClock = core clock = AHB clock.
and here we'd like to get SYSCLK (SystemCoreClock * AHB prescaler).

Reply:
I thought that  SYSCLK was an abbreviation of SystemCoreClock. I should better study STM32 RCC. I will fix it.